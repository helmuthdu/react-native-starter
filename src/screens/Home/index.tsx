import React from 'react';
import { connect } from 'react-redux';
import Home from './home.screen';
import datas from './data';
import { selectors as loadingSelectors } from '../../stores/modules/loading';

export interface Props {
  navigation: any;
  fetchList: Function;
}
export interface State {
  data: any[];
}
class HomeContainer extends React.Component<Props, State> {
  public state = {
    data: datas
  };

  public render() {
    return <Home navigation={this.props.navigation} list={this.state.data} />;
  }
}

const mapStateToProps = state => ({
  isLoading: loadingSelectors.isLoading(state)
});
export default connect(mapStateToProps)(HomeContainer);
