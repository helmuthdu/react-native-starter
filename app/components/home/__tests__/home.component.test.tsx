import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer';
import Home from '../home.component';

const navigation = { navigate: jest.fn() } as any;
const list = ['foo', 'bar'];

it('renders correctly', () => {
  const tree = renderer.create(<Home navigation={navigation} list={list} />).toJSON();
  expect(tree).toMatchSnapshot();
});
