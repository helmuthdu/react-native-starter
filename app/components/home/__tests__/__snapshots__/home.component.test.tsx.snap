// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <RCTScrollView
        contentContainerStyle={
          Object {
            "paddingTop": 30,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginBottom": 20,
                "marginTop": 10,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 80,
                  "marginLeft": -10,
                  "marginTop": 3,
                  "resizeMode": "contain",
                  "width": 100,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginHorizontal": 50,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0,0,0,0.4)",
                    "fontSize": 14,
                    "lineHeight": 19,
                    "marginBottom": 20,
                    "textAlign": "center",
                  },
                ]
              }
              uppercase={false}
            >
              Development mode is enabled, your app will be slower but you can use useful development tools.
               
              <Text
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2e78b7",
                      "fontSize": 14,
                    },
                  ]
                }
                uppercase={false}
              >
                Learn more
              </Text>
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(96,100,109, 1)",
                    "fontSize": 17,
                    "lineHeight": 24,
                    "textAlign": "center",
                  },
                ]
              }
              uppercase={false}
            >
              Get started by opening
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0,0,0,0.05)",
                    "borderRadius": 3,
                    "paddingHorizontal": 4,
                  },
                  Object {
                    "marginVertical": 7,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(96,100,109, 0.8)",
                    },
                    Object {
                      "fontFamily": "space-mono",
                    },
                  ]
                }
              >
                screens/home/home.screen.tsx
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(96,100,109, 1)",
                    "fontSize": 17,
                    "lineHeight": 24,
                    "textAlign": "center",
                  },
                ]
              }
              uppercase={false}
            >
              Change this text and your app will automatically reload.
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginTop": 15,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "paddingVertical": 15,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2e78b7",
                      "fontSize": 14,
                    },
                  ]
                }
                uppercase={false}
              >
                Help, it didnâ€™t automatically reload!
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "marginHorizontal": 15,
                "marginVertical": 15,
              }
            }
          >
            <RCTScrollView
              data={
                Array [
                  "foo",
                  "bar",
                ]
              }
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={null}
                  >
                    <View
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "borderBottomWidth": 0.5,
                          "borderColor": "#c9c9c9",
                          "flexDirection": "row",
                          "marginLeft": 16,
                          "paddingBottom": 13,
                          "paddingRight": 16,
                          "paddingTop": 13,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          }
                        }
                        uppercase={false}
                      >
                        foo
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={null}
                  >
                    <View
                      onPress={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "borderBottomWidth": 0.5,
                          "borderColor": "#c9c9c9",
                          "flexDirection": "row",
                          "marginLeft": 16,
                          "paddingBottom": 13,
                          "paddingRight": 16,
                          "paddingTop": 13,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "alignSelf": "center",
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          }
                        }
                        uppercase={false}
                      >
                        bar
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fbfbfb",
                "paddingVertical": 20,
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": -3,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 3,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(96,100,109, 1)",
                    "fontSize": 17,
                    "textAlign": "center",
                  },
                ]
              }
              uppercase={false}
            >
              This is a tab bar. You can edit it in:
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(0,0,0,0.05)",
                    "borderRadius": 3,
                    "paddingHorizontal": 4,
                  },
                  Object {
                    "marginTop": 5,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(96,100,109, 0.8)",
                    },
                    Object {
                      "fontFamily": "space-mono",
                    },
                  ]
                }
              >
                screens/index.tsx
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTScrollView>
</View>
`;
